node-commons/src/main/java/org/apache/iotdb/commons/path/fa/match/MatchedStateSet.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iotdb.commons.path.fa.match;
+
+import org.apache.iotdb.commons.path.fa.IFAState;
+
+/** This class is used for recording the states, matched by one item, in order. */
+public class MatchedStateSet {
+  private static final int INITIAL_SIZE = 8;
+
+  private final boolean[] stateStatus;
+
+  private int[] existingState = new int[INITIAL_SIZE];
+
+  private int end = 0;
+
+  MatchedStateSet(int stateSize) {
+    stateStatus = new boolean[stateSize];
+  }
+
```

Comment:
Just a suggestion for readability~

```suggestion
  /**
   * Construct an empty set with specified initial capacity.
   *
   * @param initialCapacity generally, we use stateSize as initialCapacity 
   */
  MatchedStateSet(int initialCapacity) {
    stateStatus = new boolean[initialCapacity];
  }
```